### PR TITLE
Testing the response for a simple registration message

### PR DESCRIPTION
**Pull request by LukasWoodtli**
Add a simple test for handling a registration message.

Some testing infrastructure is added that we can assert the reponse sent by Wakaama is as expected.

### CODE REVIEW

File: tests/helper/connection.c [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ *******************************************************************************/
+
+#include <assert.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "connection.h"
+#include "liblwm2m.h"
+
+#define SEND_BUFFER_MAX_LEN 100
```
Comment (mlasch) [inline tests/helper/connection.c:26]: To me the buffer size seems very small. Max packet size when using block transfer is 1024 + header. If would rather put 2048 here.

File: tests/registration_tests.c [hunk 2]
```diff
@@ -17,12 +17,75 @@
...
+    coap_free_header(&packet);
+
+    /* act */
+    lwm2m_context_t *const server_ctx = lwm2m_init(NULL);
+    lwm2m_handle_packet(server_ctx, reg_coap_msg, msg_size, NULL);
+    lwm2m_close(server_ctx);
+
+    /* assert */
+    size_t send_buffer_len;
+    uint8_t *send_buffer = test_get_send_buffer(&send_buffer_len);
```
Comment (mlasch) [inline tests/registration_tests.c:69]: If I understand this correctly this buffer then contains the registration response? At first this was not clear to me. Maybe change the variable name to stress that it is the registration response we are checking here. Send buffer is a bit misleading in my opinion.

File: tests/registration_tests.c [hunk 3]
```diff
@@ -17,12 +17,75 @@
...
+#include "er-coap-13/er-coap-13.h"
 #include "internals.h"
 #include "liblwm2m.h"
 #include "tests.h"
 
+#include <string.h>
+
 #ifdef LWM2M_SERVER_MODE
 
+static void create_test_registration_message(coap_packet_t *packet) {
```
Comment (mlasch) [inline tests/registration_tests.c:30]: This function sounds very generic but creates a very specific message for the test below. Why not having these lines directly in the test function? If we add more tests we would need a new function anyway.

Comment (LukasWoodtli) [reply tests/registration_tests.c:30]: My idea is to have, in the future, multiple helper functions that provide some testing messages for different tests.
Having this in a separate function gives it a descriptive name and make the test function a bit smaller.

**Review by mlasch — APPROVED** [review]

**Pull request merged**